### PR TITLE
Fixing custom audience batch processing

### PR DESCRIPTION
**Pull request by jshak21**
Please refer to this forum [discussion](https://twittercommunity.com/t/batch-size-for-new-audience-api/119452). Prior batch processing logic was limiting developers to 2500 users in each batch request. The 2500 limit pertains to the number of operations, not users. There is no hard limit on the number of users in each request, other than the max payload size of 5MB.

### CODE REVIEW

File: twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java [hunk 1]
```diff
@@ -253,34 +244,50 @@ public boolean hasNext() {
...
-
-                        Set<CustomAudienceUserDetails> fractionalTopOperationUsers = Sets.newHashSet();
-                        Iterator<CustomAudienceUserDetails> topOperationUsersIterator = topOperation.getUsers().iterator();
-                        while (diff > 0 && topOperationUsersIterator.hasNext()) {
-                            fractionalTopOperationUsers.add(topOperationUsersIterator.next());
-                            diff--;
+                long totalBytes = 0;
+                while (totalBytes < MAX_CUSTOM_AUDIENCE_PAYLOAD_BYTES && !operationPriorityQueue.isEmpty()) {
+                    totalBytes += CUSTOM_AUDIENCE_HEADER_BYTES;
+                    CustomAudienceOperation topOperation = operationPriorityQueue.poll();
```
Comment (symhom) [inline twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java]: nit: topOperation and fractionalTopOperation can be final unless I missed where they've been reassigned.

File: twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java [hunk 2]
```diff
@@ -253,34 +244,50 @@ public boolean hasNext() {
...
+                    CustomAudienceOperation fractionalTopOperation = new CustomAudienceOperation();
+
+                    fractionalTopOperation.setOperationType(topOperation.getOperationType());
+                    fractionalTopOperation.setEffectiveFrom(topOperation.getEffectiveFrom());
+                    fractionalTopOperation.setExpireAt(topOperation.getExpireAt());
+
+                    Set<CustomAudienceUserDetails> fractionalTopOperationUsers = Sets.newHashSet();
+                    Iterator<CustomAudienceUserDetails> topOperationUsersIterator = topOperation.getUsers().iterator();
+
+                    while (topOperationUsersIterator.hasNext()) {
```
Comment (symhom) [inline twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java:280]: I'll leave this up to you - this method seems pretty long. The while loop is a good opportunity to make it shorter but putting it in its own method.

File: twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java (hunk 1 above)
Comment (jshak21) [reply twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java]: @symhom added final keyword to variables

File: twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java (hunk 2 above)
Comment (jshak21) [reply twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java:280]: @symhom cleaned up the function a little by implementing `getSizeOfUserInBytes`. Would like to keep the while loops in the original function for simplicity.

Comment (symhom) [reply twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java:280]: Looks good!

**Pull request merged**